--- busybox-1.5.1/Makefile.clean	2007-05-20 19:00:47.000000000 +0200
+++ busybox-1.5.1/Makefile	2007-05-23 15:24:26.000000000 +0200
@@ -910,7 +910,7 @@
 
 # Directories & files removed with 'make clean'
 CLEAN_DIRS  += $(MODVERDIR)
-CLEAN_FILES +=	busybox* System.map .kernelrelease \
+CLEAN_FILES +=	busybox busybox.map busybox_unstripped System.map .kernelrelease \
                 .tmp_kallsyms* .tmp_version .tmp_busybox* .tmp_System.map
 
 # Directories & files removed with 'make mrproper'
