diff -up busybox-1.9.1/ipsvd/tcpudp.c.head busybox-1.9.1/ipsvd/tcpudp.c
--- busybox-1.9.1/ipsvd/tcpudp.c.head	2008-02-12 17:03:10.000000000 +0100
+++ busybox-1.9.1/ipsvd/tcpudp.c	2008-02-19 15:05:30.000000000 +0100
@@ -30,6 +30,9 @@
  */
 
 #include <limits.h>
+#include <linux/types.h>
+#include <netinet/in.h>
+#define u_int32_t __u32
 #include <linux/netfilter_ipv4.h> /* wants <limits.h> */
 
 #include "libbb.h"
