diff -up busybox-1.7.2/findutils/grep.c.pom busybox-1.7.2/findutils/grep.c
--- busybox-1.7.2/findutils/grep.c.pom	2007-09-03 13:48:54.000000000 +0200
+++ busybox-1.7.2/findutils/grep.c	2007-11-01 09:39:59.000000000 +0100
@@ -504,7 +504,9 @@ int grep_main(int argc, char **argv)
 			}
 		}
 		matched += grep_file(file);
+		fflush(stdout);
 		fclose_if_not_stdin(file);
+        
  grep_done: ;
 	}
 
