diff -up busybox-1.10.3/coreutils/ls.c.ls busybox-1.10.3/coreutils/ls.c
--- busybox-1.10.3/coreutils/ls.c.ls	2008-04-19 05:50:32.000000000 +0200
+++ busybox-1.10.3/coreutils/ls.c	2008-07-24 10:14:25.000000000 +0200
@@ -435,6 +435,7 @@ static void showfiles(struct dnode **dn,
 		putchar('\n');
 		column = 0;
 	}
+	fflush(stdout);
 }
 
 
@@ -478,6 +479,7 @@ static void showdirs(struct dnode **dn, 
 			}
 		}
 	}
+	fflush(stdout);
 }
 
 
