diff -up busybox-1.10.1/networking/tcpudp.c.header busybox-1.10.1/networking/tcpudp.c
--- busybox-1.10.1/networking/tcpudp.c.header	2008-04-19 06:03:13.000000000 +0200
+++ busybox-1.10.1/networking/tcpudp.c	2008-05-09 14:24:48.000000000 +0200
@@ -31,6 +31,9 @@
 
 #include "libbb.h"
 /* Wants <limits.h> etc, thus included after libbb.h: */
+#include <linux/types.h>
+#include <netinet/in.h>
+#define u_int32_t __u32 
 #include <linux/netfilter_ipv4.h>
 
 // TODO: move into this file:
