--- busybox-1.2.0/Rules.mak.ppc64	2006-07-04 13:39:13.000000000 +0200
+++ busybox-1.2.0/Rules.mak	2006-07-04 13:40:52.000000000 +0200
@@ -139,6 +139,7 @@
 HOSTCFLAGS+=$(call check_cc,$(HOSTCC),-Wstrict-prototypes,)
 HOSTCFLAGS+=$(call check_cc,$(HOSTCC),-O2,)
 HOSTCFLAGS+=$(call check_cc,$(HOSTCC),-fomit-frame-pointer,)
+HOSTCFLAGS+=$(call check_cc,$(HOSTCC),-mminimal-toc,)
 
 LD_WHOLE_ARCHIVE:=$(shell echo "int i;" > conftest.c ; \
 	$(CC) -c -o conftest.o conftest.c ; \
