diff -up busybox-1.12.1/networking/libiproute/iptunnel.c.head busybox-1.12.1/networking/libiproute/iptunnel.c
--- busybox-1.12.1/networking/libiproute/iptunnel.c.head	2008-09-28 20:04:10.000000000 +0200
+++ busybox-1.12.1/networking/libiproute/iptunnel.c	2008-12-02 15:16:44.000000000 +0100
@@ -14,13 +14,13 @@
  * Phil Karn <karn@ka9q.ampr.org>	990408:	"pmtudisc" flag
  */
 
-#include <netinet/ip.h>
 #include <net/if.h>
 #include <net/if_arp.h>
 #include <asm/types.h>
 #ifndef __constant_htons
 #define __constant_htons htons
 #endif
+#include <linux/types.h>
 #include <linux/if_tunnel.h>
 
 #include "ip_common.h"  /* #include "libbb.h" is inside */
