diff -up busybox-1.7.2/Makefile.pom busybox-1.7.2/Makefile
--- busybox-1.7.2/Makefile.pom	2007-11-06 10:21:45.000000000 +0100
+++ busybox-1.7.2/Makefile	2007-11-06 10:23:43.000000000 +0100
@@ -570,7 +570,7 @@ busybox-all  := $(core-y) $(libs-y)
 quiet_cmd_busybox__ ?= LINK    $@
       cmd_busybox__ ?= $(srctree)/scripts/trylink $(CC) $(LDFLAGS) \
       -o $@ -Wl,-Map -Wl,$@.map \
-      -Wl,--warn-common -Wl,--sort-common -Wl,--gc-sections \
+      -Wl,--warn-common -Wl,--sort-common \
       -Wl,--start-group $(busybox-all) -Wl,--end-group \
       $(LDLIBS)
 
diff -up busybox-1.7.2/applets/applets.c.pom busybox-1.7.2/applets/applets.c
--- busybox-1.7.2/applets/applets.c.pom	2007-09-16 20:48:10.000000000 +0200
+++ busybox-1.7.2/applets/applets.c	2007-11-06 10:23:20.000000000 +0100
@@ -15,18 +15,6 @@
 #include <assert.h>
 #include "busybox.h"
 
-/* Apparently uclibc defines __GLIBC__ (compat trick?). Oh well. */
-#if ENABLE_STATIC && defined(__GLIBC__) && !defined(__UCLIBC__)
-#warning Static linking against glibc produces buggy executables
-#warning (glibc does not cope well with ld --gc-sections).
-#warning See sources.redhat.com/bugzilla/show_bug.cgi?id=3400
-#warning Note that glibc is unsuitable for static linking anyway.
-#warning If you still want to do it, remove -Wl,--gc-sections
-#warning from top-level Makefile and remove this warning.
-#error Aborting compilation.
-#endif
-
-
 /* Declare <applet>_main() */
 #define PROTOTYPES
 #include "applets.h"
