diff -up busybox-1.7.2/editors/sed.c.sed busybox-1.7.2/editors/sed.c
--- busybox-1.7.2/editors/sed.c.sed	2007-09-03 13:48:55.000000000 +0200
+++ busybox-1.7.2/editors/sed.c	2007-10-31 14:41:07.000000000 +0100
@@ -1295,6 +1295,7 @@ int sed_main(int argc, char **argv)
 			bb_error_msg_and_die(bb_msg_requires_arg, "-i");
 		add_input_file(stdin);
 		process_files();
+		fflush(G.nonstdout);
 	} else {
 		int i;
 		FILE *file;
@@ -1339,8 +1340,10 @@ int sed_main(int argc, char **argv)
 			free(G.outname);
 			G.outname = 0;
 		}
-		if (G.input_file_count > G.current_input_file)
+		if (G.input_file_count > G.current_input_file) {
 			process_files();
+			fflush(G.nonstdout);
+		}
 	}
 
 	return status;
