diff -up busybox-1.10.1/networking/tcpudp.c.pom busybox-1.10.1/networking/tcpudp.c
--- busybox-1.10.1/networking/tcpudp.c.pom	2008-04-19 06:03:13.000000000 +0200
+++ busybox-1.10.1/networking/tcpudp.c	2008-05-09 12:18:53.000000000 +0200
@@ -32,6 +32,10 @@
 #include "libbb.h"
 /* Wants <limits.h> etc, thus included after libbb.h: */
 #include <linux/netfilter_ipv4.h>
+#include <linux/types.h> 
+#include <netinet/in.h> 
+#define u_int32_t __u32 
+
 
 // TODO: move into this file:
 #include "tcpudp_perhost.h"
