diff -up busybox-1.7.2/editors/sed.c.pom busybox-1.7.2/editors/sed.c
--- busybox-1.7.2/editors/sed.c.pom	2007-09-03 13:48:55.000000000 +0200
+++ busybox-1.7.2/editors/sed.c	2007-10-29 14:41:44.000000000 +0100
@@ -1339,8 +1339,10 @@ int sed_main(int argc, char **argv)
 			free(G.outname);
 			G.outname = 0;
 		}
-		if (G.input_file_count > G.current_input_file)
+		if (G.input_file_count > G.current_input_file) {
 			process_files();
+			fflush(G.nonstdout);
+		}
 	}
 
 	return status;
